core/src/elements/grenade.rs

```diff
@@ -215,7 +224,7 @@ fn update_lit_grenades(
         }
 
         // If it's time to explode
-        if grenade.age >= *fuse_time {
+        if grenade.damage_delay.elapsed_secs() >= *fuse_time {
```

Comment:
Once we set the timer duratino from the `fuse_time`, we can just do this:

```suggestion
        if grenade.damage_delay.is_finished() {
```